### ISSUE TITLE
enaGroupGet <bioproject-id> 'Completed' but no files fetched or errors

### ISSUE THREAD

**Issue by chrisgulvik** [enasequence/enaBrowserTools#19]
I'm trying the py2 version.
```
$ ./enaGroupGet PRJNA358417 2> err.log
Fetching SRR5125719
No files of format submitted for SRR5125719
Fetching SRR5125720
No files of format submitted for SRR5125720
Fetching SRR5132331
No files of format submitted for SRR5132331
Fetching SRR5132332
No files of format submitted for SRR5132332
Fetching SRR5132333
No files of format submitted for SRR5132333
Completed
$ echo  $?
0
```
./PRJNA358417/{SRR5125719,SRR5125720,SRR5132331SRR5132332,SRR5132333}/ dirs are created, but all are empty.

The stderr was empty but should list errors encountered, and exit the status shouldn't be 0 if no files were fetched.

**Comment by tseemann** [comment]
@chrisgulvik ENA stores two sets of FASTQ files, the original `submitted` ones, and the official `fastq` ones.  I've discovered that many records do not HAVE the submitted ones. If you choose `fastq` (which i think should be default because of this problem). you will get all the records.

**Comment by chrisgulvik** [comment]
Is format (output) really None?

**Comment by nicsilvester** [comment]
Not finding files for a requested format isn't an error.  Hence the information (stdout) messages and no stderr, plus the exit code.
Not all formats of submitted files are converted into FASTQ, which is why this isn't the default.  All files should have submitted files though.  If these are not showing it might be due to forking off the SRA format from the submitted files, which I still need to add into the scripts.  I'll make this a priority.

**Comment by chrisgulvik** [comment]
Ah, so the "submitted" file format, which is default, actually just means None in this case, but not all cases. I thought it meant it would fetch whatever file format(s) were submitted under the provided accession, but I suppose that's for something like `--format all`. That might be an even safer default than FastQ. I was testing a BioProject that I created, so I know they were single interleaved FastQ files submitted per isolate (to SRA not ENA).

I see here (Line 96) that because the var is set to None by default, it's just empty space printed in the stdout so that explains the message `No files of format submitted for SRR5125719` rather than `No files of format None submitted for SRR5125719`. Perhaps a more descriptive message could print something like `No files were downloaded due to weird issues with --format submitted <Default> not being able to detect which data files are available to fetch.` until there's a way to fetch `--format all`.
https://github.com/enasequence/enaBrowserTools/blob/c0911deb23679d8c25723c4fdedb85451df11810/python/readGet.py#L96

**Comment by nicsilvester** [comment]
I'll have a look into this.  It is supposed to have a default format set:
```
if output_format is None:
        output_format = utils.SUBMITTED_FORMAT
```
Note that the name is different to what is currently in the live code as we needed to rename the variable to not use a keyword.

Submitted format does mean the format submitted by users, but this is only in the case of files submitted to ENA.  All data submitted to NCBI and DDBJ are fetched from NCBI in the SRA format.  We then make FASTQ versions of each of these.  This is why you will see no file in the "submitted" option.

I'm currently working on an update that will handle COMPLETEGENOMICS_NATIVE submission format (whole directory download).  I will also plan to including a better handling for default format, an "all" option and the SRA format files into this update as well. I'm aiming to get this next release out by the end of next week.

**Comment by chrisgulvik** [comment]
Terrific, thanks for the update!

**Comment by nicsilvester** [comment]
Included in v1.4 release. Will look in the following order: submitted, sra, fastq. Note that in this case, for your examples, it will download the NCBI SRA format.

COMPLETEGENOMICS_NATIVE submission format and "all" option have been held over for the next release as I thought this change was deserving of going out without waiting for those changes.
